packages/tpetra/core/src/Tpetra_FECrsGraph_def.hpp

```diff
@@ -281,11 +286,20 @@ void FECrsGraph<LocalOrdinal, GlobalOrdinal, Node>::doOwnedPlusSharedToOwned(con
           flag[indices[j]] = true;
       }
       
-      bool lclSuccess = true;
+      bool success = true;
       for(size_t i=0; i<(size_t)flag.size(); i++)
-        if(!flag[i]) {lclSuccess=false; break;}
+        if(!flag[i]) {success=false; break;}
+
+      // Perform a reduction over the input comm, so that ranks with success=true won't be hanging.
+      // Note: this only ensures things don't hang 
+      int lclSuccess = success ? 1 : 0;
+      int gblSuccess = lclSuccess;
+      auto comm = this->getComm();
+      if (!comm.is_null()) {
+        Teuchos::reduceAll<int,int>(*comm, Teuchos::REDUCE_MIN, 1, &lclSuccess, &gblSuccess);
+      }
       TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC
-        (lclSuccess ==false,
+        (gblSuccess == 0,
```

Comment:
It's better to indicate that it is possible to disable this check.

Reply:
I'm not sure what you mean. This check is wrapped inside an if statement, and there's a comment right above the if statement, that explains that we are giving the option to bypass the check.

Reply:
> I'm not sure what you mean. This check is wrapped inside an if statement, and there's a comment right above the if statement, that explains that we are giving the option to bypass the check.

I meant in the error message. It's true that one can figure that out looking at the code. Anyway, it's not a big deal.

Reply:
Ah, I see. Done.